src/extractor/CaretValueRuleExtractor.ts

```diff
@@ -110,6 +111,11 @@ export class CaretValueRuleExtractor {
     const flatParent = getPathValuePairs(parent);
     Object.keys(flatSD).forEach(key => {
       if (flatParent[key] == null || !isEqual(flatSD[key], flatParent[key])) {
+        if (key === 'url') {
+          const existingUrl = input.url;
+          const generatedUrl = `${config.canonical}/StructureDefinition/${input.id}`;
+          if (existingUrl === generatedUrl) return;
+        }
```

Comment:
Since this check is repeated on 145-149, it may make sense to extract it out to a little helper function. I also think we could just combine everything into one `if` statement, like this:
```
if (key === 'url' && input.url === `${config.canonical}/${resourceType}/${input.id}` {
  return;
}
```
I think that will work the same, right?

Reply:
I believe it'll work the same, I'll condense this down in the next commit!